### ISSUE TITLE
gdaladdo produces artifacts with a certain set of configuration options

### ISSUE THREAD

**Issue by drnextgis** [OSGeo/gdal#4000]
If perform the following command:

```
gdaladdo -r cubic -ro --config COMPRESS_OVERVIEW DEFLATE --config INTERLEAVE_OVERVIEW PIXEL out.tif 2
```

and then open `out.tif.ovr` file you can see artifacts at the borders of the raster:

![image](https://user-images.githubusercontent.com/866124/122361752-d524df80-cf57-11eb-8250-af5f93f5887a.png)

The issue is reproducible with the latest version of GDAL 3.3.0. In GDAL 2.2.3 it works fine.

Test file: [out.tif.zip](https://github.com/OSGeo/gdal/files/6668656/out.tif.zip)

**Comment by rouault** [comment]
I've confirmed this is a duplicate of https://github.com/OSGeo/gdal/issues/3913 . Fix queued for 3.3.1

**Comment by drnextgis** [comment]
I confirm that the issue is not reproducible with `osgeo/gdal:alpine-small-latest` docker image.
